lib/api/objectPut.js

```diff
@@ -82,6 +82,10 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
                 return next(null, null);
             },
             function objectCreateAndStore(cipherBundle, next) {
+                if (objMD) {
```

Comment:
Please add tests for not overwriting data on S3 as intended and then deleting as intended.